contrib/clojure-package/examples/bert/fine-tune-bert.ipynb

```diff
@@ -0,0 +1,512 @@
+    "          tokens (bert-util/pad token-1 \"[PAD]\" seq-length)\n",
+    "        ;;; pre-processing - token to index translation\n",
+    "          indexes (bert-util/tokens->idxs token->idx tokens)]\n",
+    "    {:input-batch [indexes\n",
+    "                   token-types\n",
+    "                   [valid-length]]\n",
+    "     :label (if (= \"0\" label)\n",
+    "              [0]\n",
+    "              [1])\n",
+    "     :tokens tokens\n",
+    "     :train-item train-item}))\n",
+    "\n",
+    "(def idx->token (:idx->token vocab))\n",
+    "(def token->idx (:token->idx vocab))\n",
+    "(def dev (context/default-context))\n",
+    "(def processed-datas (mapv #(pre-processing dev idx->token token->idx %) data-train-raw))\n",
+    "(def train-count (count processed-datas))\n",
+    "(println \"Train Count is = \" train-count)\n",
+    "(println \"[PAD] token id = \" (get token->idx \"[PAD]\"))\n",
+    "(println \"[CLS] token id = \" (get token->idx \"[CLS]\"))\n",
+    "(println \"[SEP] token id = \" (get token->idx \"[SEP]\"))\n",
+    "(println \"token ids = \\n\"(-> (first processed-datas) :input-batch first)) \n",
+    "(println \"segment ids = \\n\"(-> (first processed-datas) :input-batch second)) \n",
+    "(println \"valid length = \\n\" (-> (first processed-datas) :input-batch last)) \n",
+    "(println \"label = \\n\" (-> (second processed-datas) :label)) \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have all the input-batches for each row, we are going to slice them up column-wise and create NDArray Iterators that we can use in training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "#object[org.apache.mxnet.io.NDArrayIter 0x34050a17 \"non-empty iterator\"]"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "(defn slice-inputs-data\n",
+    "  \"Each sentence pair had to be processed as a row. This breaks all\n",
+    "  the rows up into a column for creating a NDArray\"\n",
+    "  [processed-datas n]\n",
+    "  (->> processed-datas\n",
+    "       (mapv #(nth (:input-batch %) n))\n",
+    "       (flatten)\n",
+    "       (into [])))\n",
+    "\n",
+    "(def prepared-data {:data0s (slice-inputs-data processed-datas 0)\n",
+    "                    :data1s (slice-inputs-data processed-datas 1)\n",
+    "                    :data2s (slice-inputs-data processed-datas 2)\n",
+    "                    :labels (->> (mapv :label processed-datas)\n",
+    "                                 (flatten)\n",
+    "                                 (into []))\n",
+    "                    :train-num (count processed-datas)})\n",
+    "\n",
+    "(def batch-size 32)\n",
```

Comment:
Defining `batch-size` outside `train-data` is unnecessary, as it is `let`-bound on line 380.